datagen/ridership/process.py

```diff
@@ -15,62 +23,140 @@
+
     final = (
-        final.groupby(["year", "week", "route_or_line"])["validations"]
+        final.groupby(["year", "week", route_key])[count_key]
         .mean()
         .round()
         .reset_index()
     )
 
     final = final.merge(dates, on=["week", "year"], how="left")
 
-    # get list of bus routes
-    routelist = list(set(final["route_or_line"].tolist()))
+    # get list of routes
+    routelist = list(set(final[route_key].tolist()))
 
     # create dict
     output = {}
 
     # write out each set of routes to dict
     for route in routelist:
-        dftemp = final[final["route_or_line"] == route]
-        dictdata = (
-            dftemp[["servicedate", "validations"]]
-            .rename(columns={"servicedate": "date", "validations": "riders"})
-            .to_dict(orient="records")
-        )
-        output[route] = dictdata
-
+        for_route = final[final[route_key] == route]
+        only_date_and_count = for_route[[date_key, count_key]].dropna()
+        dictdata = only_date_and_count.rename(
+            columns={date_key: "date", count_key: "riders"}
+        ).to_dict(orient="records")
+        route_id = route_ids_map[route] if route_ids_map else route
+        output[route_id] = dictdata
     return output
 
 
+def estimate_daily_boardings_by_cr_route(path_to_csv_file: str):
+    df = pd.read_csv(path_to_csv_file)
+    df = df[(df["season"] == "Spring 2018") & (df["day_type_name"] == "weekday")]
+    df = df[["route_id", "average_ons"]]
+    merged = (
+        df.groupby(df["route_id"])
+        .aggregate({"route_id": "first", "average_ons": "sum"})
+        .rename(columns={"average_ons": "boardings"})
+    )
+    sum_boardings = merged["boardings"].sum()
+    merged["scaled_boardings"] = (
+        merged["boardings"] * (CR_RIDERSHIP_BASELINE / sum_boardings)
+    ).astype(int)
+    out_dict = {}
+    for entry in merged.to_dict(orient="records"):
+        out_dict[entry["route_id"]] = entry["scaled_boardings"]
+    # We lose a few when flooring to integers but...
+    assert abs(sum(out_dict.values()) - CR_RIDERSHIP_BASELINE) < 10
+    return out_dict
+
+
+def format_subway_data(path_to_csv_file: str):
+    return format_ridership_csv(
+        path_to_csv_file=path_to_csv_file,
+        date_key="servicedate",
+        route_key="route_or_line",
+        count_key="validations",
+    )
+
+
+def format_cr_data(path_to_ridershp_file: str, path_to_seasonal_ridership_file: str):
+    estimations = estimate_daily_boardings_by_cr_route(path_to_seasonal_ridership_file)
+    ridership_by_route = format_ridership_csv(
+        path_to_csv_file=path_to_ridershp_file,
+        date_key="service_date",
+        route_key="line",
+        count_key="estimated_boardings",
+        route_ids_map=unofficial_cr_labels_map,
+    )
+    with_estimations = {}
+    for route_id, route_dates in ridership_by_route.items():
+        estimate = estimations[route_id]
+        with_estimations[route_id] = [
+            {
```

Comment:
This is our hack to create a baseline on the charts: we fill in our inferred baseline for day 1, and then on day 2 we add a `0` value. Downstream code will fill in `0` for March-June 2020. (We could consider adding something to the chart here).